pipeline-scripts/buildlib.groovy

```diff
@@ -297,13 +330,63 @@ def build_puddle(conf_url, keys, Object...args) {
     return puddle_dir
 }
 
+
 def with_virtualenv(path, f) {
     final env = [
         "VIRTUAL_ENV=${path}",
         "PATH=${path}/bin:${env.PATH}",
         "PYTHON_HOME=",
     ]
     return withEnv(env, f)
+
+def parse_record_log( working_dir ) {
+    def record = readFile( "${working_dir}/record.log" )
+    lines = record.split("\\r?\\n");
+
+    def result = [:]
+    int i = 0
+    int f = 0
+    for ( i = 0; i < lines.length; i++ ) {
+        fields = lines[i].split("|")
+        type = fields[0]
+        if(! result.containsKey(type)) {
+            result[type] = []
+        }
+        record = [:]
+        for ( f = 1; f = fields.length; f++ ){
```

Comment:
f < fields.length . This script should be testable in a local jenkins instance. Just setup a normal Jenkins pipeline and paste your code in. Prepopulate a record.log on the file system and work out the kinks.